zerver/lib/markdown/__init__.py

```diff
@@ -2191,12 +2191,12 @@ def build_inlinepatterns(self) -> markdown.util.Registry:
         # Custom strikethrough syntax: ~~foo~~
         DEL_RE = r"(?<!~)(\~\~)([^~\n]+?)(\~\~)(?!~)"
         # Custom bold syntax: **foo** but not __foo__
-        # str inside ** must start and end with a word character
-        # it need for things like "const char *x = (char *)y"
-        EMPHASIS_RE = r"(\*)(?!\s+)([^\*^\n]+)(?<!\s)\*"
-        STRONG_RE = r"(\*\*)([^\n]+?)\2"
-        STRONG_EM_RE = r"(\*\*\*)(?!\s+)([^\*^\n]+)(?<!\s)\*\*\*"
-        TEX_RE = r"\B(?<!\$)\$\$(?P<body>[^\n_$](\\\$|[^$\n])*)\$\$(?!\$)\B"
+        # String inside ** must start and end with a word
+        # character; this makes us CommonMark compliant.
```

Comment:
Our emphasis implementation is nowhere close to CommonMark compliant. CommonMark specifies a series of [17 rules](https://spec.commonmark.org/0.30/#emphasis-and-strong-emphasis) for parsing emphasis, accompanied by 131 examples and a [16 step detailed algorithm](https://spec.commonmark.org/0.30/#process-emphasis). We should not pretend that our three dinky regexes correctly implement that all.

Reply:
@andersk I didn't actually mean that we are fully CommonMark compliant in this case.
I didn't find the original comment to convey the actual reason why we want the string to start and end with `**`.